instrumentation-api-incubator/src/main/java/io/opentelemetry/instrumentation/api/incubator/builder/internal/DefaultHttpServerInstrumenterBuilder.java

```diff
@@ -0,0 +1,230 @@
+      List<String> responseHeaders) {
+    httpAttributesExtractorBuilder.setCapturedResponseHeaders(responseHeaders);
+    return this;
+  }
+
+  /**
+   * Configures the instrumentation to recognize an alternative set of HTTP request methods.
+   *
+   * <p>By default, this instrumentation defines "known" methods as the ones listed in <a
+   * href="https://www.rfc-editor.org/rfc/rfc9110.html#name-methods">RFC9110</a> and the PATCH
+   * method defined in <a href="https://www.rfc-editor.org/rfc/rfc5789.html">RFC5789</a>.
+   *
+   * <p>Note: calling this method <b>overrides</b> the default known method sets completely; it does
+   * not supplement it.
+   *
+   * @param knownMethods A set of recognized HTTP request methods.
+   * @see HttpServerAttributesExtractorBuilder#setKnownMethods(Set)
+   */
+  @CanIgnoreReturnValue
+  public DefaultHttpServerInstrumenterBuilder<REQUEST, RESPONSE> setKnownMethods(
+      Set<String> knownMethods) {
+    httpAttributesExtractorBuilder.setKnownMethods(knownMethods);
+    httpSpanNameExtractorBuilder.setKnownMethods(knownMethods);
+    httpServerRouteBuilder.setKnownMethods(knownMethods);
+    return this;
+  }
+
+  @CanIgnoreReturnValue
+  public DefaultHttpServerInstrumenterBuilder<REQUEST, RESPONSE> setHeaderGetter(
+      @Nullable TextMapGetter<REQUEST> headerGetter) {
+    this.headerGetter = headerGetter;
+    return this;
+  }
+
+  /**
+   * Configures the instrumentation to emit experimental HTTP server metrics.
+   *
+   * @param emitExperimentalHttpServerMetrics {@code true} if the experimental HTTP server metrics
+   *     are to be emitted.
+   */
+  @CanIgnoreReturnValue
+  public DefaultHttpServerInstrumenterBuilder<REQUEST, RESPONSE>
+      setEmitExperimentalHttpServerMetrics(boolean emitExperimentalHttpServerMetrics) {
+    this.emitExperimentalHttpServerMetrics = emitExperimentalHttpServerMetrics;
+    return this;
+  }
+
+  /** Sets custom {@link SpanNameExtractor} via transform function. */
+  @CanIgnoreReturnValue
+  public DefaultHttpServerInstrumenterBuilder<REQUEST, RESPONSE> setSpanNameExtractor(
+      Function<SpanNameExtractor<REQUEST>, ? extends SpanNameExtractor<? super REQUEST>>
+          spanNameExtractorTransformer) {
+    this.spanNameExtractorTransformer = spanNameExtractorTransformer;
+    return this;
+  }
+
+  public Instrumenter<REQUEST, RESPONSE> build() {
+    return build(b -> {});
+  }
+
+  public Instrumenter<REQUEST, RESPONSE> build(
+      Consumer<InstrumenterBuilder<REQUEST, RESPONSE>> instrumenterBuilderConsumer) {
+
+    InstrumenterBuilder<REQUEST, RESPONSE> builder = builder();
+    instrumenterBuilderConsumer.accept(builder);
+
+    if (headerGetter != null) {
+      return builder.buildServerInstrumenter(headerGetter);
+    }
+    return builder.buildInstrumenter(SpanKindExtractor.alwaysServer());
+  }
+
+  public InstrumenterBuilder<REQUEST, RESPONSE> builder() {
```

Comment:
it looks like this is only used externally by tests, is there a way to change those tests to use same API as other usage patterns?
```suggestion
  private InstrumenterBuilder<REQUEST, RESPONSE> builder() {
```

Reply:
I think it's not - intellij finds groovy files that are false positives